app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,77 @@
+class SubmissionsController < ApplicationController
+  before_action :authenticate_user!, only: [:create, :create_comment]
+  before_action :authenticate_to_submit, only: :new
+
+  helper_method :posts, :post, :new_post, :new_comment
+
+  def new
+  end
+
+  def show
+  end
+
+  def index
+  end
+
+  def create
+    @post = Submission.new(post_params)
+    @post.user_id = current_user.id
+    if @post.save
+      flash[:alert] = "Your submission was saved"
+      redirect_to submission_path(@post.id)
+    else
+      flash[:alert] = @post.errors.messages
+      redirect_to new_submission_path
+    end
+  end
+
+  def create_comment
+    @comment = Submission.new(comment_params)
+    @comment.user_id = current_user.id
+    @comment.post_id = parent_post_id
```

Comment:
instead of setting the attributes, the params hash could be modified to include them. I'm thinking something like:

```ruby
@comment = current_user.submissions.build(comment_params.merge(post_id: parent_post_id))
```

Could also modify the `comment_params` method to merge in the `post_id`. Your call.

Reply:
Nice - I forgot about merge, I was trying to use `(comments_params, {user_id: ..., etc })` and getting a too many arguments error

Reply:
Also, this helps resolve some confusion I've been having about the relationship between the `new_comment` method and this `create_comment`method.... I think you may have made the issue explicit in your comment on the `new_post` method.

I needed the `new_post` and `new_comment` methods to support the blank form fields on the GET, but the #create actions also have `Submission.new`. I didn't think I needed `Submission.new` twice per post or comment, but wasn't sure where to have it.